searchsorted to work with uint type

After the Numpy 2.0 transition search_sorted no longer works for `uint64` dtype.  `uint64` was removed from `test_searchsorted` because it was causing an error:

```
self = <arkouda.client.ZmqChannel object at 0x7e101ce377e0>
cmd = 'searchSorted<float64,1,1>', recv_binary = False
args = '["{\\"key\\": \\"x1\\", \\"dtype\\": \\"float64\\", \\"val\\": \\"id_cxxDjvW_19\\"}", "{\\"key\\": \\"x2\\", \\"dtype...uint64\\", \\"val\\": \\"id_cxxDjvW_20\\"}", "{\\"key\\": \\"side\\", \\"dtype\\": \\"str\\", \\"val\\": \\"left\\"}"]'
size = 3, request_id = None

    def send_string_message(
        self,
        cmd: str,
        recv_binary: bool = False,
        args: Optional[str] = None,
        size: int = -1,
        request_id: Optional[str] = None,
    ) -> Union[str, memoryview]:
        message = RequestMessage(
            user=username, token=self.token, cmd=cmd, format=MessageFormat.STRING, args=args, size=size
        )
        # Note - Size is a placeholder here because Binary msg not yet support json args and
        # request_id is a noop for now
        logger.debug(f"sending message {json.dumps(message.asdict())}")
    
        self.socket.send_string(json.dumps(message.asdict()))
    
        if recv_binary:
            frame = self.socket.recv(copy=False)
            view = frame.buffer
            # raise errors sent back from the server
            if bytes(view[0 : len(b"Error:")]) == b"Error:":
                raise RuntimeError(frame.bytes.decode())
            return view
        else:
            raw_message = self.socket.recv_string()
            try:
                return_message = ReplyMessage.fromdict(json.loads(raw_message))
    
                # raise errors or warnings sent back from the server
                if return_message.msgType == MessageType.ERROR:
>                   raise RuntimeError(return_message.msg)
E                   RuntimeError: Error executing command: cannot cast class to type - runtime types not compatible

../../arkouda/client.py:532: RuntimeError
```

Fix this and add `uint64` back into `test_searchsorted`.

--- a/arkouda/numpy/sorting.py
+++ b/arkouda/numpy/sorting.py
@@ -324,6 +324,7 @@ def searchsorted(
         If `a` has more than one dimension.
     TypeError
         If `a` has an unsupported dtype (i.e., not int64, uint64, bigint, or float64).
+        If the dtype of `a` and `v` does not match
 
 
     Examples
@@ -352,6 +353,9 @@ def searchsorted(
     else:
         scalar_input = True
         v_ = cast(pdarray, array([v]))
+
+    if a.dtype != v_.dtype:
+        raise TypeError(f"The dtype of a ({a.dtype}) and v ({v_.dtype}) must match.")
 
     repMsg = generic_msg(
         cmd=f"searchSorted<{a.dtype},{a.ndim},{v_.ndim}>",

--- a/tests/numpy/sort_test.py
+++ b/tests/numpy/sort_test.py
@@ -131,7 +131,7 @@ class TestSort:
             assert np.allclose(np.sort(npa), ak.sort(ak.array(npa), algo).to_ndarray(), equal_nan=True)
 
     @pytest.mark.parametrize("size", pytest.prob_size)
-    @pytest.mark.parametrize("dtype", [ak.float64, ak.int64, ak.bigint])
+    @pytest.mark.parametrize("dtype", [ak.float64, ak.int64, ak.bigint, ak.uint64])
     @pytest.mark.parametrize("v_shape", [(), (10,), (4, 5), (2, 2, 3)])
     @pytest.mark.parametrize("side", ["left", "right"])
     def test_searchsorted(self, size, dtype, v_shape, side):
@@ -154,10 +154,10 @@ class TestSort:
                 pytest.skip(f"Server not compiled for rank {len(v_shape)}")
             else:
                 v = ak.randint(low=low, high=high, size=v_shape, dtype=dtype_, seed=pytest.seed)
-                v = ak.array(v, dtype) + shift
+                v = ak.array(ak.array(v, dtype) + shift, dtype)
         a = ak.randint(low=low, high=high, size=size, dtype=dtype_, seed=pytest.seed)
         a = ak.sort(a)
-        a = ak.array(a, dtype) + shift
+        a = ak.array(ak.array(a, dtype) + shift, dtype)
         np_a = a.to_ndarray()
         if isinstance(v, ak.pdarray):
             np_v = v.to_ndarray()

